### PR TITLE
Add support for interactive partitioning

### PR DESCRIPTION
**Pull request by poncovka**
Add DBus API that can be used in the custom partitioning spoke.

### CODE REVIEW

**Comment by poncovka** [comment]
Fixed some small issues.

File: tests/nosetests/pyanaconda_tests/module_scheduler_test.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+        report = self.interface.ValidateContainerName("_my/contain$er")
+        self._check_report(report, "Invalid container name. Try 'my_container'.")
+
+        report = self.interface.ValidateContainerName("my_container")
+        self._check_report(report, None)
+
+    def validate_raid_level_test(self):
+        """Test ValidateRaidLevel."""
+        report = self.interface.ValidateRaidLevel("raid6", 2)
+        self._check_report(report, "The RAID level you have selected raid6 requires more "
```
Comment (VladimirSlavik) [inline tests/nosetests/pyanaconda_tests/module_scheduler_test.py]: Not actually related to this PR, but the "raid6" part is missing some quotes, brackets or the like, I think?

Comment (poncovka) [reply tests/nosetests/pyanaconda_tests/module_scheduler_test.py]: Fixed.

**Comment by poncovka** [comment]
Rebased.

**Comment by poncovka** [comment]
The failing test is not related to this PR.

File: pyanaconda/modules/storage/partitioning/interactive/utils.py [hunk 2]
```diff
@@ -340,6 +340,21 @@ def validate_mount_point(path, mount_points):
...
+
+    :param storage: an instance of Blivet
+    :param name: a container name
+    :return: an error message or None
+    """
+    safe_name = storage.safe_device_name(name)
+
+    if name != safe_name:
+        return _("Invalid container name. Try '{}'.").format(safe_name)
+
```
Comment (jkonecny12) [inline pyanaconda/modules/storage/partitioning/interactive/utils.py]: This calls for a method in blivet which would be named `validate_device_name`, this construct seems really weird.

File: pyanaconda/modules/storage/partitioning/interactive/utils.py [hunk 3]
```diff
@@ -668,19 +668,19 @@ def collect_file_system_types(device):
     """
     # Collect the supported filesystem types.
     supported_types = {
-        fs.name for fs in get_supported_filesystems()
-        if fs.name not in UNSUPPORTED_FILESYSTEMS
+        fs.type for fs in get_supported_filesystems()
+        if fs.type not in UNSUPPORTED_FILESYSTEMS
```
Comment (jkonecny12) [inline pyanaconda/modules/storage/partitioning/interactive/utils.py]: You ale switching from names to types but in ``:return:`` is talking about ``..system names``

File: pyanaconda/modules/storage/partitioning/interactive/scheduler_interface.py [hunk 4]
```diff
@@ -35,3 +36,13 @@ def GenerateSystemName(self) -> Str:
         :return: a translated string
         """
         return self.implementation.generate_system_name()
+
+    def GenerateSystemData(self, boot_drive: Str) -> Structure:
+        """Generate the new installation data.
+
+        :param boot_drive: a name of the boot drive
+        :return: an instance of OSData
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/modules/storage/partitioning/interactive/scheduler_interface.py]: Fixed.

File: pyanaconda/modules/storage/partitioning/interactive/utils.py (hunk 2 above)
Comment (poncovka) [reply pyanaconda/modules/storage/partitioning/interactive/utils.py]: Fixed.

File: pyanaconda/modules/storage/partitioning/interactive/utils.py (hunk 3 above)
Comment (poncovka) [reply pyanaconda/modules/storage/partitioning/interactive/utils.py]: Fixed.

**Comment by poncovka** [comment]
> Shouldn't the new submodule be named `planner` instead of `scheduler`? The `scheduler` word invokes me more time scheduling than planning partitioning.

I don't think so. It follows the Blivet's terminology. We are scheduling actions.

**Comment by poncovka** [comment]
Rebased and fixed some small issues.

**Comment by jkonecny12** [comment]
> > Shouldn't the new submodule be named `planner` instead of `scheduler`? The `scheduler` word invokes me more time scheduling than planning partitioning.
> 
> I don't think so. It follows the Blivet's terminology. We are scheduling actions.

I'm more inclined that we are planning storage and they are scheduling actions. That action creation and execution is more covered in deeps of Blivet.

**Pull request merged**